### ISSUE TITLE
[Core] Dynamic Node Labeling and Resource Registration 

### ISSUE THREAD

**Issue by Irvingwangjr** [ray-project/ray#45375]
### Description

right now, Ray provides node labeling and resource registration when calling ‘ray start --label/--custom-resource‘ or using RAY_OVERRIDE_RESOURCES/RAY_OVERRIDE_LABELS env var. As the Ray Cluster's lifecycle become longer, we hope we can label the node dynamic, and apply nodeLabelSelector to dispatch the actors or tasks

**Comment by rynewang** [comment]
What's your use case for dynamic labels? I think today you can create a new node with new labels, and shut down the old node.

**Comment by Irvingwangjr** [comment]
yeah, if the workload is submitted using RayJob, using an ephemeral Ray Cluster, we can add the custom label at starting. But if we have a long-running Ray Cluster, that won't work.

**Comment by Irvingwangjr** [comment]
if we want to implement this, this can be achieved by using RaySyncer? 
Syncer will trigger NodeManager, do ConsumeSyncMessage -> UpdateResourceUsage -> UpdateNode -> ClusterResourceManager.AddOrUpdateNode. eventually update NodeResources data Structure (which contains labels)

**Comment by Daraan** [comment]
This would be a great feature. Especially on clusters where nodes have lifetimes like SLURM one could set a label like *ending-soon* before timeout and exclude this label from any scheduling.

This is very crucial for entrypoint jobs which have to stay alive and we not want to start them on nodes that terminate in the near future.

**Comment by zhangsikai123** [comment]
Looking forward to this feature, our workaround for this feature is simply create multiple worker groups including all existing labels, but this makes cluster initiation very cumbersome and difficult.
